dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeMetadataConnector.java

```diff
@@ -148,16 +155,30 @@ protected void addSqlTasksWithInfoSchemaFallback(
     }
   }
 
-  @ForOverride
-  protected void addSingleSqlTask(
+  private void addSingleSqlTask(
       @Nonnull List<? super Task<?>> out,
-      @Nonnull Class<? extends Enum<?>> header,
       @Nonnull String format,
-      @Nonnull TaskVariant task) {
+      @Nonnull TaskVariant task,
+      @Nonnull Function<ResultSet, String[]> transformer) {
     out.add(
         new JdbcSelectTask(
                 task.zipEntryName, String.format(format, task.schemaName, task.whereClause))
-            .withHeaderClass(header));
+            .withHeaderTransformer(transformer));
+  }
+
+  @CheckReturnValue
```

Comment:
Did this mean @CheckForNull? And did that go away once we make the exception/interface change?

Reply:
Yup, it went away with the new design